src/provider/states/download_package.rs

```diff
@@ -0,0 +1,137 @@
+use std::path::{Path, PathBuf};
```

Comment:
Would it make sense to call the files the same as the state?
download_package.rs -> Downloading